From e662f817e61235c3b35b9237af69c960351a7949 Mon Sep 17 00:00:00 2001
From: anonymous <anonymous@inventec.com>
Date: Wed, 12 Aug 2020 23:43:28 -0400
Subject: [PATCH 1/3] Modified sel file location path and add RecordID property

- Changed sel log path to "/var/lib/sel_log"
- Add DBUS property "RecordID" to check the current record index
---
 include/sel_logger.hpp |  3 ++-
 src/sel_logger.cpp     | 20 ++++++++++++++++++++
 2 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/include/sel_logger.hpp b/include/sel_logger.hpp
index 35f8b95..89328bb 100644
--- a/include/sel_logger.hpp
+++ b/include/sel_logger.hpp
@@ -34,7 +34,8 @@ static constexpr size_t selEvtDataMaxSize = 3;
 static constexpr size_t selOemDataMaxSize = 13;
 static constexpr uint8_t selEvtDataUnspecified = 0xFF;
 
-static const std::filesystem::path selLogDir = "/var/log";
+unsigned int recordIndex = 0;
+static const std::filesystem::path selLogDir = "/var/lib/sel_log";
 static const std::string selLogFilename = "ipmi_sel";
 
 template <typename... T>
diff --git a/src/sel_logger.cpp b/src/sel_logger.cpp
index 83d487a..9596e81 100644
--- a/src/sel_logger.cpp
+++ b/src/sel_logger.cpp
@@ -129,6 +129,7 @@ static unsigned int getNewRecordId(void)
     {
         recordId = 1;
     }
+    recordIndex = recordId;
     return recordId;
 }
 #endif
@@ -235,6 +236,25 @@ int main(int argc, char* argv[])
            const uint8_t& recordType) {
             return selAddOemRecord(message, selData, recordType);
         });
+    // Add RecordID property to check the current record index
+    ifaceAddSel->register_property("RecordID", recordIndex,
+                                   // set handle
+                                   [&](const unsigned int& record, 
+                                       unsigned int& propertyValue) {
+                                       if (record < selInvalidRecID)
+                                       {
+                                           recordIndex = record;
+                                       }
+
+                                       return 1;
+                                   },
+                                   // get handle
+                                   [&](unsigned int& propertyValue) {
+                                       propertyValue = recordIndex;
+                                       return propertyValue;
+                                   });
+    recordIndex = initializeRecordId();
+
     ifaceAddSel->initialize();
 
 #ifdef SEL_LOGGER_MONITOR_THRESHOLD_EVENTS
-- 
2.7.4

